test/powershell/Language/Classes/scripting.enums.tests.ps1

```diff
@@ -70,6 +70,50 @@ Describe 'enums' -Tags "CI" {
         It 'E5 has correct value' { [E5]::e0 | Should -Be ([E5]40) }
         It 'E6 has correct value' { [E6]::e0 | Should -Be ([E6]38) }
     }
+
+    Context 'Enum with non-default underlying type' {
+        enum EX1 : byte { A;B;C;D }
+        enum EX2 : sbyte { A;B;C;D }
+        enum EX3 : short { A;B;C;D }
+        enum EX4 : ushort { A;B;C;D }
+        enum EX5 : int { A;B;C;D }
+        enum EX6 : uint { A;B;C;D }
+        enum EX7 : long { A;B;C;D }
+        enum EX8 : ulong { A;B;C;D }
+
+        It 'EX1 has the specified underlying type' { [Enum]::GetUnderlyingType([EX1]) | Should -Be ([byte]) }
+        It 'EX2 has the specified underlying type' { [Enum]::GetUnderlyingType([EX2]) | Should -Be ([sbyte]) }
+        It 'EX3 has the specified underlying type' { [Enum]::GetUnderlyingType([EX3]) | Should -Be ([short]) }
+        It 'EX4 has the specified underlying type' { [Enum]::GetUnderlyingType([EX4]) | Should -Be ([ushort]) }
+        It 'EX5 has the specified underlying type' { [Enum]::GetUnderlyingType([EX5]) | Should -Be ([int]) }
+        It 'EX6 has the specified underlying type' { [Enum]::GetUnderlyingType([EX6]) | Should -Be ([uint]) }
+        It 'EX7 has the specified underlying type' { [Enum]::GetUnderlyingType([EX7]) | Should -Be ([long]) }
+        It 'EX8 has the specified underlying type' { [Enum]::GetUnderlyingType([EX8]) | Should -Be ([ulong]) }
+    }
+
+    Context 'Enum with FlagsAttribute' {
+        [Flags()]
+        enum BitMask { A;B;C;D }
+
+        It 'Consecutive member values double when FlagsAttribute is present' {
+            [BitMask]::A.value__ | Should -Be 0
+            [BitMask]::B.value__ | Should -Be 1
+            [BitMask]::C.value__ | Should -Be 2
+            [BitMask]::D.value__ | Should -Be 4
+        }
+    }
+
+    Context 'Enum with negative user-specified values' {
+        enum V1 {
+            A = -4
+            B
+        }
```

Comment:
From line 41. This should works:
```powershell
enum V1 {
    A = -4
    B = [int]::MinValue
    C = -1
    D
}
```

Reply:
Difference is that we're counting _towards_ zero when the previous value is negative, so the test on line 41 doesn't make much sense in relation to `Int.MinValue`. 

That being said, we should accept `Int.MinValue`, so I added a test for that + next consecutive auto-assignment